tree/ntuple/v7/src/RField.cxx

```diff
@@ -2191,6 +2188,93 @@ void ROOT::Experimental::RArrayField::AcceptVisitor(Detail::RFieldVisitor &visit
    visitor.VisitArrayField(*this);
 }
 
+//------------------------------------------------------------------------------
+// RArrayAsRVecField
+
+ROOT::Experimental::RArrayAsRVecField::RArrayAsRVecField(
+   std::string_view fieldName, std::unique_ptr<ROOT::Experimental::Detail::RFieldBase> itemField,
+   std::size_t arrayLength)
+   : ROOT::Experimental::Detail::RFieldBase(fieldName, "ROOT::VecOps::RVec<" + itemField->GetType() + ">",
+                                            ENTupleStructure::kCollection, false /* isSimple */, arrayLength),
+     fItemSize(itemField->GetValueSize()),
+     fArrayLength(arrayLength)
+{
+   Attach(std::move(itemField));
```

Comment:
I think we should also add a check for the structural role an number of repetitions of `itemField`, as in
```suggestion
   R__ASSERT(itemField->GetStructure() == kLeaf && itemField->GetNRepetitions() >= 1);
   Attach(std::move(itemField));
```

Reply:
Piggy-backing on my comment from above, this assert would almost always fail, as it is not at all guaranteed that `itemField->GetNRepetitions() >= 1`. On the contrary, most often it is not the case. It is only the case if we are dealing with a nested collection such as `std::array<std::array>`